### PR TITLE
Roadmap update july 2024

### PR DESCRIPTION
**Pull request by tracymiranda**
#### Summary
Progresses #472 - updates Sigstore Roadmap, specifically ecosystem adoption and language client updates

#### Release Note
NONE

#### Documentation
N/A

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -48,8 +48,8 @@ Sigstore has several implementations of its core signing and verification workfl
 |      Javascript      | [sigstore/sigstore-js](https://github.com/sigstore/sigstore-js)     |         GA        |
 |         Java         | [sigstore/sigstore-java](https://github.com/sigstore/sigstore-java)   |        Beta       |
 |         Rust         | [sigstore/sigstore-rs](https://github.com/sigstore/sigstore-rs)     |        Beta       |
-|        Go            | [sigstore/sigstore-go](https://github.com/sigstore/sigstore-go)     | Under development |
-|         Ruby         | [sigstore/ruby-sigstore](https://github.com/sigstore/ruby-sigstore)   |     pre-Alpha     |
+|        Go            | [sigstore/sigstore-go](https://github.com/sigstore/sigstore-go)     | Beta |
+|         Ruby         | [sigstore/sigstore-ruby](https://github.com/sigstore/sigstore-ruby)   |     pre-Alpha     |
```
Comment (haydentherapper) [inline ROADMAP.md:52]: I would call this alpha

File: ROADMAP.md [hunk 2]
```diff
@@ -71,9 +71,23 @@ A package manager looking to adopt Sigstore as part of its artifact signing and
 
 The npm registry, the world's largest open source package manager for the JavaScript ecosystem, [recently announced built-in support for Sigstore-signed SLSA build provenance](https://github.blog/2023-04-19-introducing-npm-package-provenance/) to users. This allows consumers of packages from npm to independently verify the link between a package fetched from the npm registry to the source code from which it was built. The build provenance attestation is cryptographically signed using Sigstore. Other package managers may find the [documentation of npm's Sigstore adoption and resulting design decisions](https://repos.openssf.org/build-provenance-for-all-package-registries) useful.
```
Comment (haydentherapper) [inline ROADMAP.md:73]: Can we add a paragraph about GitHub artifact attestations before Homebrew, and then mention homebrew leverages that?

File: ROADMAP.md [hunk 3]
```diff
@@ -71,9 +71,23 @@ A package manager looking to adopt Sigstore as part of its artifact signing and
...
+
 The Python ecosystem (and its official third party package repository PyPI) are also [working on integrating Sigstore](https://yossarian.net/res/pub/pycon-2023.pdf) into their package publishing and verification workflow via PEPs 458, 480, 691 & 694 based on the recent sigstore-python GA release. 
 
-The Java ecosystem is also working to integrate Sigstore, through a Java-native SDK and integrations with [Maven Central](https://blog.sonatype.com/maven-central-and-sigstore). The Sigstore community is also working closely with the Eclipse Foundation who is exploring adding support for signing and verification of artifacts within Eclipse projects.
```
Comment (haydentherapper) [inline ROADMAP.md:76]: Can we mention Pep740, which is ongoing work for PyPI adoption?

**Review by bobcallaway — DISMISSED** [review]

**Comment by tracymiranda** [comment]
I've used separate commits for clarity, though these and merge branch commits should be squashed on commit (or if preference to squash now let me know and I can do so)

**Review by haydentherapper — APPROVED** [review]

**Review by bobcallaway — APPROVED** [review]

**Pull request merged**